language/e2e-tests/src/tests/failed_transaction_tests.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{account::AccountData, executor::FakeExecutor};
+use libra_types::vm_error::{StatusCode, VMStatus};
+use vm::{
+    gas_schedule::{GasAlgebra, GasPrice, GasUnits},
+    transaction_metadata::TransactionMetadata,
+};
+use vm_runtime::{data_cache::BlockDataCache, LibraVM};
+
+#[test]
+fn failed_transaction_cleanup_test() {
+    let mut fake_executor = FakeExecutor::from_genesis_file();
+    let sender = AccountData::new(1_000_000, 10);
+    fake_executor.add_account_data(&sender);
+
+    let mut libra_vm = LibraVM::new(fake_executor.config());
+    let mut data_cache = BlockDataCache::new(fake_executor.get_state_view());
+    libra_vm.load_gas_schedule(&data_cache);
```

Comment:
It might be nicer to provide a c`fg(test)` method on `LibraVM` that does all of this initialization for you given a `VMConfig` (e.g., `LibraVM::from(config)`). That would also let us keep `load_gas_schedule` private.

Reply:
I'll just keep it as it is for now as we will probably revise the logic here anyway in the coming future as the gas schedule will become part of the on-chain configuration story.